log/sequencer.go

```diff
@@ -169,12 +169,15 @@ func (s Sequencer) buildMerkleTreeFromStorageAtRoot(ctx context.Context, root *t
 	return mt, nil
 }
 
-func (s Sequencer) buildNodesFromNodeMap(nodeMap map[string]storage.Node, newVersion int64) ([]storage.Node, error) {
+func (s Sequencer) buildNodesFromNodeMap(nodeMap map[compact.NodeID][]byte, newVersion int64) ([]storage.Node, error) {
```

Comment:
What's the win from changing string -> n.NodeID - is it mainly removing the cost of stringification?

Reply:
- Removing stringification cost (time+memory+allocations)
- Using human readable strings as IDs is a slightly worse practice
- Simpler `updateCompactTree` (no error handling), and `storage.NodeID` is pushed closer to the place where it is needed
- In the long term it would be great to use integer node IDs in logs even at the storage layer (again less overhead and error handling)